mava/systems/parameter_client.py

```diff
@@ -110,6 +102,22 @@ def _adjust_and_request(self) -> None:
         )
         self._copy(self._client.get_parameters(self._get_keys))
 
+    def _async_adjust_and_request(self) -> futures.Future:
+        """Set the parameters in the server, then update local params from the server.
+
+        Returns:
+            A future for the parameter get
+        """
+        self._client.futures.set_parameters(  # type: ignore
+            {key: self._parameters[key] for key in self._set_keys},
+        )
+        # not sure how these futures will interact
+        # will it fetch the parameters set above?
+        get_future = self._client.futures.get_parameters(self._get_keys)  # type: ignore
+        get_future.add_done_callback(lambda ctx: self._copy(ctx.result()))
```

Comment:
This could cause a race condition so I'm not sure what to do here? 

A possibility is to only copy the stuff in get keys, because if we are setting we should have the most recent params relating to set keys right?

Any suggestions welcome :smile:

Reply:
Fixed potential race condition by only getting parameters not in set_keys